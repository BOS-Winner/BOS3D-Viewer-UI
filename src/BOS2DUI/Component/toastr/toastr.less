:global{
  #toast-container{
    position: absolute;
    .toast{
      border-radius: 4px;
      opacity:1;
      padding: 10px 10px 10px 40px;
    }

    >.toast-error{
      color: #333333;
      border: 1px solid #F03D3D;
      background-color: #FFF1F1;
      background-image:url('./img/error.png')!important;
    }
    >.toast-success{
      color: #333333;
      border: 1px solid #3ED3A3;
      background-color: #E8FFEB;
      background-image:url('./img/success.png')!important;
    }
    >.toast-info{
      color: #333333;
      border: 1px solid #2878FF;
      background-color: #FFFFFF;
      background-image:url('./img/info.png')!important;
    }
    >.toast-warning{
      color: #333333;
      border: 1px solid #FFAD0D;
      background-color: #FFFAEB;
      background-image:url('./img/warning.png')!important;
    }
  }
}

