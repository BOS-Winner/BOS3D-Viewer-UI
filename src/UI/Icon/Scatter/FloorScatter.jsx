import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
// import Icon from "../../Base/Icon";
// // import scatterPng from "../img/white/floorScatter.png";
// import style from "./style.less";
// import { AntdIcon } from '../../utils/utils';
// import iconstyle from '../../Toolbar/bottom.less';

/**
 * 楼层分解
 */
class ModelScatter extends React.Component {
  static propTypes = {
    viewer: PropTypes.object.isRequired,
    enable: PropTypes.bool.isRequired,
    coefficient: PropTypes.number.isRequired,
  };

  shouldComponentUpdate(nextProps) {
    if (nextProps.enable && nextProps.coefficient !== this.props.coefficient) {
      this.onRangeChange(nextProps.coefficient);
      return false;
    }
    return true;
  }

  componentDidUpdate(prevProps) {
    if (prevProps.enable && !this.props.enable) {
      const viewer3D = this.props.viewer;
      viewer3D.closeFloorExplosion();
    }
  }

  onRangeChange(value) {
    const coefficient = parseFloat(value);
    const viewer3D = this.props.viewer;
    viewer3D.floorExplosion(coefficient);
  }

  render() {
    return (
      <div
        // className={style.rangeContainer}
        title="楼层分解"
      >
        {/* <Icon
          icon={<AntdIcon type="iconloucengfenjie-01" className={iconstyle.icon} />}
          title="楼层"
          showTitle={false}
        /> */}
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  viewer: state.system.viewer3D,
});
const mapDispatchToProps = () => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ModelScatter);
