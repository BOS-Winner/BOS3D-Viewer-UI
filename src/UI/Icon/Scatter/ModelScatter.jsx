import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
// import Icon from "Base/Icon";
// // import scatterPng from "../img/white/modelScatter.png";
// import style from "./style.less";
// import { AntdIcon } from '../../utils/utils';
// import iconstyle from '../../Toolbar/bottom.less';

/**
 * 模型分解
 */
class ModelScatter extends React.Component {
  static propTypes = {
    viewer: PropTypes.object.isRequired,
    enable: PropTypes.bool.isRequired,
    coefficient: PropTypes.number.isRequired,
  };

  shouldComponentUpdate(nextProps) {
    if (nextProps.enable && nextProps.coefficient !== this.props.coefficient) {
      this.onRangeChange(nextProps.coefficient);
      return false;
    }
    return true;
  }

  componentDidUpdate(prevProps) {
    if (prevProps.enable && !this.props.enable) {
      const viewer3D = this.props.viewer;
      const modelKeys = viewer3D.getViewerImpl().modelManager.getModelKeys();
      viewer3D.closeModelsExplosion(modelKeys);
    }
  }

  onRangeChange(value) {
    const coefficient = parseFloat(value);
    const viewer3D = this.props.viewer;
    const modelKeys = viewer3D.getViewerImpl().modelManager.getModelKeys();
    viewer3D.modelsExplosion({
      modelKey: modelKeys,
      coefficientX: coefficient,
      coefficientY: coefficient,
      coefficientZ: coefficient,
    });
  }

  render() {
    return (
      <div
        // className={style.rangeContainer}
        title="模型分解"
      >

        {/* <Icon
          icon={<AntdIcon type="icondecompose" className={iconstyle.icon} />}
          title="模型"
          showTitle={false}
        /> */}
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  viewer: state.system.viewer3D,
});
const mapDispatchToProps = () => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ModelScatter);
