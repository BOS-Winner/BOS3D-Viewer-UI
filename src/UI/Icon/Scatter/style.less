.popup{
  display: flex;
  flex-direction: column;
  position: absolute;
  bottom: 0;
}

.range-container {
  position: relative;
}

.scatter-group{
  width: 100%;
  // height: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.slider-container{
  margin: 16px 0 0 0;
  .slider-wrapper{
    margin: 25px 0 0 0;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    >span{
      width: 16px;
      color: rgba(187, 187, 187, 1);
    }
    .range{
      // flex: 4;
      width: 100%;
      :global{
        .bos3d-slider-rail{
          background-color: rgba(68, 68, 68, 1);
        }
        .bos3d-slider-track{
          background: rgba(40, 120, 255, 1);
        }
        .bos3d-slider-handle{
          @handleSize: 8px;
          width: @handleSize;
          height: @handleSize;
          margin: -2px 0 0 0;
        }
      }
    }
  }
}