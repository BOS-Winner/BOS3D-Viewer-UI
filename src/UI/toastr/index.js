/**
 * 提供默认的toastr配置
 * 文档&样例 https://codeseven.github.io/toastr/demo.html
 */
import "toastr/build/toastr.min.css";
import toastr from "toastr";
import "./toastr.less";

toastr.options = {
  "closeButton": false,
  "progressBar": false,
  "positionClass": "toast-top-center",
  "showDuration": 300,
  "hideDuration": 300,
  "timeOut": 3000,
  "extendedTimeOut": 1000,
  "preventDuplicates": true,
  "showEasing": "swing",
  "hideEasing": "linear",
  "showMethod": "fadeIn",
  "hideMethod": "fadeOut",
};

export default toastr;
