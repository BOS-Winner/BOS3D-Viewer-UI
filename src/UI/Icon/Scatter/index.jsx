import React from 'react';
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Slider } from 'antd';
import Icon from "Base/Icon";
import Modal from 'Base/Modal';
import Card from 'Base/Card';
import ModelScatter from "./ModelScatter.jsx";
import FloorScatter from "./FloorScatter.jsx";
import style from "./style.less";
import toastr from "../../toastr";
import { AntdIcon } from '../../utils/utils';
import iconstyle from '../../Toolbar/bottom.less';
import { DEFAULT_MODAL_PLACE } from "../../constant";

class Scatter extends React.Component {
  static propTypes = {
    viewer: PropTypes.object.isRequired,
    mode: PropTypes.string.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      // showPopup: false,
      enableModelScatter: false,
      enableFloorScatter: false,
      coefficient: 1,
      showModal: false,
    };
  }

  componentDidUpdate(prevProps) {
    if (prevProps.mode !== this.props.mode && this.props.mode === "漫游模式") {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({
        // showPopup: false,
        enableModelScatter: false,
        enableFloorScatter: false,
        coefficient: 1,
        showModal: false,
      });
    }
  }

  triggerPopUp = () => {
    // ev.preventDefault();
    // ev.stopPropagation();
    const { enableFloorScatter, enableModelScatter } = this.state;
    // console.log("打开modal");
    if (enableFloorScatter || enableModelScatter) {
      this.setState({
        enableModelScatter: false,
        enableFloorScatter: false,
        coefficient: 1,
        showModal: false
      });
    } else {
      this.setState(state => ({
        showModal: !state.showModal,
      }));
    }
  }

  onRangeChange = (value) => {
    this.setState({
      coefficient: parseFloat(value)
    });
  }

  switchScatter(type) {
    if (type === 'model') {
      this.setState(state => ({
        enableModelScatter: !state.enableModelScatter,
        enableFloorScatter: false,
      }));
    }
    if (type === 'floor') {
      if (this.props.viewer.canFloorExplosion()) {
        this.setState(state => ({
          enableModelScatter: false,
          enableFloorScatter: !state.enableFloorScatter,
        }));
      } else {
        toastr.info("该模型不支持楼层分解", "", {
          target: `#${this.props.viewer.viewport}`,
        });
      }
    }
    this.onRangeChange(1);
  }

  render() {
    const {
      enableFloorScatter, enableModelScatter, coefficient, showModal
    } = this.state;

    // 分解模式
    const scatterList = [
      {
        icon: 'icondecompose',
        title: '模型分解',
        func: () => this.switchScatter('model'),
        active: enableModelScatter
      },
      {
        icon: 'iconloucengfenjie-01',
        title: '楼层分解',
        func: () => this.switchScatter('floor'),
        active: enableFloorScatter
      }
    ];
    // 滑动条属性
    const sliderProps = {
      min: 1,
      max: 3,
      tipFormatter: null,
      defaultValue: 1,
      step: 0.01,
      disabled: !(enableFloorScatter || enableModelScatter)
    };

    return (
      <div>
        <Modal
          visible={showModal}
          onCancel={this.triggerPopUp}
          title="分解"
          width="350px"
          height="245px"
          top={DEFAULT_MODAL_PLACE.scatter.top}
          right={DEFAULT_MODAL_PLACE.scatter.right}
          left={DEFAULT_MODAL_PLACE.scatter.left}
          minWidth={250}
          minHeight={245}
          viewportDiv={this.props.viewer.viewportDiv}
        >
          <div className={style.scatterGroup}>
            {scatterList.map(
              item => (
                <Card
                  key={item.title}
                  icon={item.icon}
                  title={item.title}
                  func={item.func}
                  active={item.active}
                />
              ))}
          </div>
          <div className={style.sliderContainer}>
            <span>
              离散系数:
              {coefficient}
            </span>
            {/* 调用子组件 */}
            <div>
              <ModelScatter
                enable={enableModelScatter}
                coefficient={coefficient}
              />
              <FloorScatter
                enable={enableFloorScatter}
                coefficient={coefficient}
              />
            </div>
            <div className={style.sliderWrapper}>
              <span>1</span>
              <Slider
                {...sliderProps}
                className={style.range}
                onChange={this.onRangeChange}
                value={coefficient}
              />
              <span>3</span>
            </div>
          </div>
        </Modal>
        <div
          role="button"
          tabIndex={0}
          title="分解"
          onClick={ev => this.triggerPopUp(ev)}
        >
          <Icon
            icon={<AntdIcon type="icongongnengmianban_fenjie-01" className={iconstyle.icon} />}
            selected={showModal}
            title="分解"
          />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  viewer: state.system.viewer3D,
  mode: state.button.mode,
});
const mapDispatchToProps = () => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Scatter);
