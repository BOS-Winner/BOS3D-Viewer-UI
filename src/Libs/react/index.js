import React from "react";
import useUpdateEffect from "./useUpdateEffect";

React.useUpdateEffect = useUpdateEffect;

export default React;
// export { default as useUpdateEffect } from "./useUpdateEffect";
