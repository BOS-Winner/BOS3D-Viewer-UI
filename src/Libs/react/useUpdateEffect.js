import { useEffect, useRef } from "react";

export default function (effect, deps) {
  const isFirstRender = useRef(true);
  useEffect(() => {
    if (isFirstRender.current) {
      isFirstRender.current = false;
    } else {
      effect();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, deps);
}
