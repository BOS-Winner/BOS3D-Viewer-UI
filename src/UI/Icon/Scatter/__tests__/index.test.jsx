import React from 'react';
import { render, act } from '@testing-library/react';
import { Provider } from 'react-redux';
import { genStore } from 'jest/initTest';
import Scatter from '../index';
import toastr from '../../../toastr';

jest.mock('../../../toastr', () => ({
  info: jest.fn(),
}));

describe('scatter index', () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should switch scatter mode', () => {
    const store = genStore();
    const { container, getByTitle } = render((
      <Provider store={store}>
        <Scatter />
      </Provider>
    ));
    // 初始状态
    expect(container).toMatchSnapshot();

    act(() => {
      getByTitle('分解').click();
    });
    // 弹出选项
    expect(container).toMatchSnapshot();

    act(() => {
      getByTitle('模型分解').click();
    });
    // 收起选项，选中index分解按钮
    expect(container).toMatchSnapshot();

    act(() => {
      getByTitle('分解').click();
    });
    // 回到初始
    expect(container).toMatchSnapshot();

    act(() => {
      getByTitle('分解').click();
    });
    act(() => {
      getByTitle('楼层分解').click();
    });
    // 收起选项，选中index分解按钮
    expect(container).toMatchSnapshot();

    store.getState().system.viewer3D.canFloorExplosion = () => false;
    act(() => {
      getByTitle('分解').click();
      getByTitle('分解').click();
    });
    act(() => {
      getByTitle('楼层分解').click();
    });
    expect(toastr.info).toBeCalledTimes(1);
    expect(toastr.info.mock.calls[0]).toMatchSnapshot();
  });
});
