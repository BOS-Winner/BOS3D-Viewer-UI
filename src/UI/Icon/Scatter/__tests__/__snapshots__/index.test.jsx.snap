// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scatter index should switch scatter mode 1`] = `
<div>
  <div>
    <div
      style="position: relative;"
    >
      <div
        class="toolbar popup"
        style="display: none;"
      >
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="模型分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(modelScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="楼层分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(floorScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        class="range"
        role="presentation"
        style="display: none;"
      >
        <span>
          离散系数：
          1
        </span>
        <div
          class="container"
          role="presentation"
        >
          <div
            class="rail"
          />
          <div
            class="track"
            style="width: 0%;"
          />
          <div
            class="step"
            style="left: 0%;"
          />
        </div>
      </div>
    </div>
    <div
      role="button"
      tabindex="0"
      title="分解"
    >
      <div
        class="icon "
        data-selected="false"
        role="button"
        style="background-image: url(scatter.png);"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`scatter index should switch scatter mode 2`] = `
<div>
  <div>
    <div
      style="position: relative;"
    >
      <div
        class="toolbar popup"
        style="display: flex;"
      >
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="模型分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(modelScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="楼层分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(floorScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        class="range"
        role="presentation"
        style="display: none;"
      >
        <span>
          离散系数：
          1
        </span>
        <div
          class="container"
          role="presentation"
        >
          <div
            class="rail"
          />
          <div
            class="track"
            style="width: 0%;"
          />
          <div
            class="step"
            style="left: 0%;"
          />
        </div>
      </div>
    </div>
    <div
      role="button"
      tabindex="0"
      title="分解"
    >
      <div
        class="icon "
        data-selected="false"
        role="button"
        style="background-image: url(scatter.png);"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`scatter index should switch scatter mode 3`] = `
<div>
  <div>
    <div
      style="position: relative;"
    >
      <div
        class="toolbar popup"
        style="display: none;"
      >
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="模型分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(modelScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="楼层分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(floorScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        class="range"
        role="presentation"
        style=""
      >
        <span>
          离散系数：
          1
        </span>
        <div
          class="container"
          role="presentation"
        >
          <div
            class="rail"
          />
          <div
            class="track"
            style="width: 0%;"
          />
          <div
            class="step"
            style="left: 0%;"
          />
        </div>
      </div>
    </div>
    <div
      role="button"
      tabindex="0"
      title="分解"
    >
      <div
        class="icon "
        data-selected="true"
        role="button"
        style="background-image: url(scatter.png);"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`scatter index should switch scatter mode 4`] = `
<div>
  <div>
    <div
      style="position: relative;"
    >
      <div
        class="toolbar popup"
        style="display: none;"
      >
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="模型分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(modelScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="楼层分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(floorScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        class="range"
        role="presentation"
        style="display: none;"
      >
        <span>
          离散系数：
          1
        </span>
        <div
          class="container"
          role="presentation"
        >
          <div
            class="rail"
          />
          <div
            class="track"
            style="width: 0%;"
          />
          <div
            class="step"
            style="left: 0%;"
          />
        </div>
      </div>
    </div>
    <div
      role="button"
      tabindex="0"
      title="分解"
    >
      <div
        class="icon "
        data-selected="false"
        role="button"
        style="background-image: url(scatter.png);"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`scatter index should switch scatter mode 5`] = `
<div>
  <div>
    <div
      style="position: relative;"
    >
      <div
        class="toolbar popup"
        style="display: none;"
      >
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="模型分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(modelScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
        <div
          role="button"
          tabindex="0"
        >
          <div
            class="rangeContainer"
            title="楼层分解"
          >
            <div
              class="icon "
              data-selected="false"
              role="button"
              style="background-image: url(floorScatter.png);"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        class="range"
        role="presentation"
        style=""
      >
        <span>
          离散系数：
          1
        </span>
        <div
          class="container"
          role="presentation"
        >
          <div
            class="rail"
          />
          <div
            class="track"
            style="width: 0%;"
          />
          <div
            class="step"
            style="left: 0%;"
          />
        </div>
      </div>
    </div>
    <div
      role="button"
      tabindex="0"
      title="分解"
    >
      <div
        class="icon "
        data-selected="true"
        role="button"
        style="background-image: url(scatter.png);"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`scatter index should switch scatter mode 6`] = `
Array [
  "该模型不支持楼层分解",
  "",
  Object {
    "target": "#id",
  },
]
`;
